src/summaries/chapter5.yaml

```diff
@@ -0,0 +1,136 @@
+    
+    Syntax
+      ```
+      solidity
+        contract <Derived Contract Name> is <Base Contract Name>{
+      }
+      ```
+  
+    Solidity provides different types of inheritance.
+      - Single Inheritance
+        * In Single or single-level inheritance, the functions, and variables of one base contract are inherited to only one derived contract.
+        
+      - Multi-level Inheritance
+        * Very similar to single inheritance.
+        * Instead of just a single parent-child relationship, there are multiple levels of the parent-child relationship.
+  
+      - Hierarchical Inheritance
+        * Here, a single contract acts as a base contract for multiple derived contracts.
+        * It is mostly used when a common functionality is used in different places.
+  
+      - Multiple Inheritance
+        * Here, a single contract can be inherited from one or more contracts.
+      
+    Note : Solidity follows the path of Python and uses C3 Linearization, also known as Method Resolution Order (MRO), to force a specific order in graphs of base contracts. The order of execution is determined by order of the declaration at the contract level. It has nothing to do with the order you call/pass arguments up from the constructor's signature.
+ 
+- title: Polymorphism in Solidity
+  shortTitle: Polymorphism
+  key: polymorphism
+  details: |
+    It is an ability to process data in more than one form.
+  
+    Solidity supports two types of polymorphism.
+      - Function polymorphism
+        * It is also known as method overloading.
+        * Multiple functions are declared to have the same name within the same contract or inheriting the contract.
+        * Functions differ based on the number of parameters or parameter datatypes.
+        * Declaration of function cannot be overloaded by functions that differ only in return type.
+      
+      - Contract polymorphism
+        * Using multiple contract instances interchangeably when they are related to each other by using inheritance.
+ 
+- title: Constructor for Solidity
+  shortTitle: Constructor
+  key: constructor
+  details: |
+    It is a special method in any object-oriented programming language that gets called whenever an object of a class is initialized.
+    Solidity provides a constructor declaration inside the smart contract, and it invokes only once when the contract is deployed and is used to initialize the contract state.
+    The compiler creates a default constructor if there is no explicitly defined constructor.
+  
+    Syntax
+      ```
+      solidity
+      constructor() <Access Modifier> {         
+      }
+      ```
+  
+    Note
+      * It is very useful in a smart contract, a parameter value can be defined at the run time and can also restrict the method call.
+      * Constructor overloading is not supported in Solidity; it only allows one constructor at a time.
+ 
+- title: Abstraction in Solidity
```

Comment:
We need to also cover libraries and interfaces

https://www.tutorialspoint.com/solidity/solidity_interfaces.htm
https://www.tutorialspoint.com/solidity/solidity_libraries.htm